src/main/java/blusunrize/immersiveengineering/api/shader/ShaderRegistry.java

```diff
@@ -29,12 +29,12 @@
 	/**
 	 * A map of shader name to ShaderRegistryEntry, which contains ShaderCases, rarity, weight and loot specifics
 	 */
-	public static LinkedHashMap<String, ShaderRegistryEntry> shaderRegistry = new LinkedHashMap<>();
+	public static LinkedHashMap<ResourceLocation, ShaderRegistryEntry> shaderRegistry = new LinkedHashMap<>();
```

Comment:
Changing all of these fields to the "basic" data structure (`Map<...>`, `List<...>`, etc) would probably be a good idea.

Reply:
Only on the left, I presume?

Reply:
Yes. How would you even change it on the RHS?!